src/framework/mlt_property.c

```diff
@@ -1322,6 +1328,11 @@ static void refresh_animation( mlt_property self, double fps, locale_t locale, i
 		self->animation = mlt_animation_new();
 		self->serialiser = (mlt_serialiser) mlt_animation_serialize_tf;
 		mlt_animation_parse( self->animation, self->prop_string, length, fps, locale );
+		// Delete the string and interact with the animation from now on.
+		self->types &= !mlt_prop_string;
```

Comment:
I think this needs to be bitwise not operator instead of logical not.
https://en.wikipedia.org/wiki/Bitwise_operations_in_C#Bitwise_NOT_~_/_ones'_complement_(unary)

Reply:
Good catch. Will fix if we go with this method.